File: arl-backend/app/schemas/export.py
"""Export schemas for request/response validation"""

from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from enum import Enum


class ExportFormat(str, Enum):
    """Export format types"""
    JSON = "json"
    MARKDOWN = "markdown"
    HTML = "html"
    PDF = "pdf"


class ExportStatus(str, Enum):
    """Export job status"""
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"


class ExportConfig(BaseModel):
    """Export configuration options"""
    include_code: bool = True
    include_outputs: bool = True
    include_visualizations: bool = True


class ExportCreate(BaseModel):
    """Create export request"""
    project_id: str
    format: ExportFormat
    include_code: Optional[bool] = True
    include_outputs: Optional[bool] = True
    include_visualizations: Optional[bool] = True


class ExportResponse(BaseModel):
    """Export response schema"""
    id: str
    project_id: str
    format: ExportFormat
    status: ExportStatus
    created_at: datetime
    completed_at: Optional[datetime] = None
    download_url: Optional[str] = None
    error: Optional[str] = None

    class Config:
        from_attributes = True

File: arl-frontend/src/types/export.ts
export type ExportFormat = 'json' | 'markdown' | 'html' | 'pdf';

export type ExportStatus = 'pending' | 'processing' | 'completed' | 'failed';

export interface ExportRequest {
  project_id: string;
  format: ExportFormat;
  include_code?: boolean;
  include_outputs?: boolean;
  include_visualizations?: boolean;
}

export interface ExportJob {
  id: string;
  project_id: string;
  format: ExportFormat;
  status: ExportStatus;
  created_at: string;
  completed_at?: string;
  download_url?: string;
  error?: string;
}
